torch/_export/utils.py

```diff
@@ -475,9 +509,21 @@ def _extract_pytree_key(x):
     name_map: Dict[str, str] = {}
 
     # map user input names with mod.forward() signature
-    combined_args = (
-        inspect.signature(mod.forward).bind(*fake_args, **fake_kwargs).arguments
+    sig = (
+        inspect.signature(mod.forward)
+        if not _is_torch_jit_trace
+        else _get_torch_jit_trace_forward_signature(mod)
```

Comment:
Hmm i am not really in favor of having branching logic here in general. Can we somehow wrap this logic somewhere so we don't have to use this flag? 
```
def get_signature(mod):
      if isinstance(mod, torch.jit.ScriptModule, torch.jit.TracedModule):
          _get_torch_jit_trace_forward_signature
      else:
          inspect.signature(mod.forward)
```
Even better would be TS-d modules being compatible with inspect.signature if possible? Maybe we can leave TODO for now.

Reply:
Sure, I will fix the logic.

I had gave it some thoughts of making TS-d modules being compatible directly with inspect.signature, which requires access to the original forward function object. One potential way to get that object is to get the object from forward source code with `compile` and `(TS-d module).code`. However, it will be very buggy to do (e.g., the standalone function has some global variable reference or import some new package, etc). So, I feel the most reliable way is to go from AST, which only does symbolic analysis of the function source code. Any thoughts on this @angelayi? 

I will leave a TODO for now.

Reply:
If it is not that easy it's ok

Reply:
I think it's fine to use the AST. I think Tugsuu was just commented on separating the branching to a helper function.